### DISCUSSION TITLE
How to set visibility visibility    to  "moleculer-db" mixin registers  actions

### DISCUSSION THREAD

**Discussion by sachinda123456** [moleculerjs/moleculer#1016] [General]
I need to know if there any way to set visibility level   to moleculer-db  default operations such as

 /**
     *  - list
     *  - find
     *  - count
     *  - create
     *  - insert
     *  - update
     *  - remove
     */

**Comment by avasiliev83** [comment]
This question is too actual for me to.
Please help!!!

**Comment by ccampanale** [comment]
I have not tested this myself but my understanding is that you can provide an additional service mixin to modify the DB adapter mixin actions. This would work similar to the ability to disable these actions altogether as documented [here](https://moleculer.services/docs/0.14/services.html#Merge-algorithm).

Example setting actions to `private` using `moleculer-db` readme example:

```js
"use strict";

const { ServiceBroker } = require("moleculer");
const DbService = require("moleculer-db");

const broker = new ServiceBroker();

broker.createService({
    name: "users",
    mixins: [
        DbService,
        {
            actions: {
                list: { visibility: 'private' },
                find: { visibility: 'private' },
                count: { visibility: 'private' },
                create: { visibility: 'private' },
                insert: { visibility: 'private' },
                update: { visibility: 'private' },
                remove: { visibility: 'private' },
            }
        }
    ],

    settings: {
        fields: ["_id", "username", "name"]
    },

    afterConnected() {
        // Seed the DB with ˙this.create`
    }
});

broker.start()
```
